fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryInParameterJoinPlan.java

```diff
@@ -105,4 +111,37 @@ public void logPlanStructure(StoreTimer timer) {
         timer.increment(FDBStoreTimer.Counts.PLAN_IN_PARAMETER);
         getInner().logPlanStructure(timer);
     }
+
+    @Nonnull
+    @Override
+    public InternalPlannerGraph rewriteInternalPlannerGraph(@Nonnull List<InternalPlannerGraph> childGraphs) {
+        final InternalPlannerGraph.Node root =
+                new InternalPlannerGraph.Node(this,
+                        getClass().getSimpleName());
+        final InternalPlannerGraph graphForInner = Iterables.getOnlyElement(childGraphs);
+        final InternalPlannerGraph.SourceNode explodeNode = new InternalPlannerGraph.SourceNode("Explode", externalBinding);
+        final InternalPlannerGraph.Edge fromExplodeEdge = new InternalPlannerGraph.Edge();
+        return InternalPlannerGraph.builder(root)
+                .addGraph(graphForInner)
+                .addNode(explodeNode)
+                .addEdge(explodeNode, root, fromExplodeEdge)
+                .addEdge(graphForInner.getRoot(), root, new InternalPlannerGraph.Edge(ImmutableSet.of(fromExplodeEdge)))
+                .build();
+    }
+
+    @Nonnull
+    @Override
+    public ExplainPlannerGraph rewriteExplainPlannerGraph(@Nonnull List<ExplainPlannerGraph> childGraphs) {
```

Comment:
The amount of code duplication between `rewriteInternalPlannerGraph()` and `rewriteExplainPlannerGraph()` is quite unfortunate. I assume it's tricky to make this better because of the type system? Unless we can think of a fundamental reason to change it, we should open an issue about cleaning this up (but I won't block the merge on resolving this).

This close relationship seems to hold for (almost?) all of the implementations of these methods, which does suggest that something is awry with our abstractions.

Reply:
I have redesigned this for better code reuse in a way that we can allow for a specific deviation from the common path if explain is different than internal show. If they are the same both use cases now do use the same node and edge hierarchy and can therefore reuse large amounts of code. I am happier with the that than to keep classes strictly separated but also duplicated.